plugins/item_tasks/server/rest.py

```diff
@@ -168,6 +168,30 @@ def _transformOutputs(self, outputs, token, job):
 
         return transformed
 
+    def _validateOutputParentType(self, outputId, parentType, outputSpec):
+        """
+        Checks if the output parent type is compatible with the output type.
+        """
+
+        # Find the corresponding output specification for the given outputID
+        output = None
+        for out in outputSpec:
+            if outputId == out['id']:
+                output = out
+                break
+
+        # If a corresponding output is found, check if its parent type is valid
+        if output:
```

Comment:
Python `for` loops take an optional `else:` clause, which only runs if the loop iterates to completion without any `break` being triggered. You could use this to eliminate both sides of the `if output:` conditional, moving the contained logic up one level.